### DISCUSSION TITLE
Pyscript doesn't recognise library

### DISCUSSION THREAD

**Discussion by rjvbeek** [custom-components/pyscript#215] [Q&A] [answered]
Hi,

I'm playing around with some triggers based on data from my PSN account (like online status), using the PSNAWP library. I have a script which runs fine locally, runs fine via the HASS terminal (SSH add-on), but it throws an error when I make it into a service:

Exception in <file.psn.psn_availability> line 3: from psnawp_api import psnawp ^ ModuleNotFoundError: No module named 'psnawp_api'

It looks like pyscript cannot find the installed library, any ideas on why this is failing?

Thanks in advance!

**Comment by rjvbeek** [comment, marked as answer]
I managed to fix this by adding the library PLUS VERSION in the pyscript/requirements.txt file. I tried various values in there before it finally worked. Running into another issue now, but I'll try to solve that myself first ;)

**Reply by gonenutsbrb** [reply]
You have, no idea, how long I was looking for this answer.  I tried RTFM like 50 times and didn't seem to understand this piece of it.

Thank you. Thank you. Thank you.

Especially thank you for coming back and [saying how you fixed it](https://xkcd.com/979/).  I owe you, like so many beers.
